tests/observability/metrics/conftest.py

```diff
@@ -1046,3 +1050,23 @@ def deleted_ssp_operator_pod(admin_client, hco_namespace):
 @pytest.fixture(scope="class")
 def initiate_metric_value(request, prometheus):
     return get_metrics_value(prometheus=prometheus, metrics_name=request.param)
+
+
+@pytest.fixture(scope="class")
+def vnic_info_from_vm_and_vmi(vm_for_test):
+    vmi_instance = vm_for_test.vmi.instance
+    vm_instance = vm_for_test.instance
+    vm_instance_binding_name_and_type = binding_name_and_type_from_vm_or_vmi(vm=vm_instance)
+    vmi_instance_binding_name_and_type = binding_name_and_type_from_vm_or_vmi(vm=vmi_instance)
+    return {
+        "vm": {
+            VNIC_NAME: vm_instance.spec.template.spec.networks[0].name,
+            BINDING_NAME: vm_instance_binding_name_and_type[BINDING_NAME],
+            BINDING_TYPE: vm_instance_binding_name_and_type[BINDING_TYPE],
+        },
+        "vmi": {
+            VNIC_NAME: vmi_instance.spec.networks[0].name,
+            BINDING_NAME: vmi_instance_binding_name_and_type[BINDING_NAME],
+            BINDING_TYPE: vmi_instance_binding_name_and_type[BINDING_TYPE],
+        },
+    }
```

Comment:
this should be parametrized via request and based on that return either vm info or vmi info